### PR TITLE
redirect to new docset

### PR DESCRIPTION
**Pull request by terakilobyte**
## Pull Request Info

This redirects drivers/go to drivers/go/current/ and into the new docset.

### Issue JIRA link:
N/A

### Snooty build log:
https://workerpool-boxgs.mongodbstitch.com/pages/job.html?jobId=6144eca1c608bec57f3c4fa7

### Docs staging link (requires sign-in on MongoDB Corp SSO):
https://docs-mongodborg-staging.corp.mongodb.com/ecosystem/docsworker/NNNNN/index.html

### Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [ ] Did you run a spell-check?
- [ ] Did you run a grammar-check?
- [ ] Does it render on staging correctly?
- [ ] Are all the links working?
- [ ] Are the staging and workerpool job links in the PR description updated?

### CODE REVIEW

**Review by ccho-mongodb — APPROVED** [review]
LGTM

File: config/redirects [hunk 1]
```diff
@@ -4,6 +4,9 @@ define: manual https://docs.mongodb.com/manual
 define: atlas https://docs.atlas.mongodb.com
 define: devhub https://developer.mongodb.com
 
+# Go
+raw: drivers/go -> ${base}/go/current/
```
Comment (biniona-mongodb) [inline config/redirects:8]: Question: Any reason why this redirect goes to the current branch? Seems like Java and [Node](https://github.com/mongodb/docs-ecosystem/blob/b78f0518081ab9ffcf176a323c335ec488500974/config/redirects#L57) don't go to a branch?

Comment (ccho-mongodb) [reply config/redirects:8]: > Question: Any reason why this redirect goes to the current branch? Seems like Java and [Node](https://github.com/mongodb/docs-ecosystem/blob/b78f0518081ab9ffcf176a323c335ec488500974/config/redirects#L57) don't go to a branch?

There's no redirect for Java (java/sync) in this file. We added a different redirect to address how we renamed the drawer that contains all the JVM drivers.

The `drivers/javascript` redirect was likely added when we only had one version of node. We should update that one to current as it currently leads to two redirects (drivers/javascript -> drivers/node -> drivers/node/current).

The Java one is not a redirect to the java/sync driver.

**Comment by kyuan-mongodb** [comment]
Question: Don't you also need to update the TOC here like you did for the [Java docs](https://github.com/mongodb/docs-ecosystem/pull/749/files)?

**Comment by terakilobyte** [comment]
@ccho-mongodb Good point about the node redirect, I'll add that to this. 
@kyuan-mongodb There's no need since the redirect will capture it.

**Comment by ccho-mongodb** [comment]
> Question: Don't you also need to update the TOC here like you did for the [Java docs](https://github.com/mongodb/docs-ecosystem/pull/749/files)?

Yes, I think we should update the index ToC to avoid an unnecessary redirect:
```
Go Driver </go/current/>
```

**Pull request merged**